akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpEntity.scala

```diff
@@ -605,19 +605,21 @@ object HttpEntity {
           // "no limit"
           case Some(SizeLimit(bytes, cl @ Some(contentLength))) ⇒
             if (contentLength > bytes) throw EntityStreamSizeException(bytes, cl)
-          // else we still count but never throw an error
+            maxBytes = bytes
+            bytesLeft = bytes
           case Some(SizeLimit(bytes, None)) ⇒
             maxBytes = bytes
             bytesLeft = bytes
           case None ⇒
+          // else we still count but never throw an error
         }
       }
 
       override def onPush(): Unit = {
         val elem = grab(in)
         bytesLeft -= sizeOf(elem)
         if (bytesLeft >= 0) push(out, elem)
-        else failStage(EntityStreamSizeException(maxBytes))
+        else failStage(EntityStreamSizeException(maxBytes, Some(maxBytes - bytesLeft)))
```

Comment:
ah, well I got confused. And of course this stream needs to fail. I somehow was already inside the Http2ServerDemuxer...